lib/deserializers.php

```diff
@@ -161,3 +161,31 @@ function elementList(Reader $reader, $namespace = null) {
     return $values;
 
 }
+
+/**
+ * @param Reader $reader
+ * @param $popo
+ * @param $namespace
+ * @return null|$popo
+ */
+function popoObject(Reader $reader, $popo, $namespace) {
+    if ($reader->isEmptyElement) {
+        $reader->next();
+        return null;
+    }
+
+    $reader->read();
+    do {
+
+        if ($reader->nodeType === Reader::ELEMENT && $reader->namespaceURI == $namespace) {
```

Comment:
Yeah, I also thought about such things... I dont like annotations that much because they add a magic layer which needs to be documented very properly. Also it mixes the serialize/deserialize logic into the ValueObjects which is the opposite direction this PR tries to aim for.

Maybe this will improve in php-src future in case annotations get a 1st class citizen in php.